### PR TITLE
[pvr] use user-defined recording sorts

### PR DESCRIPTION
**Pull request by emveepee**
## Description

Currently PVR Recordings are sorted by a hard-coded, hidden values SortByDate and SortOrderDescending.  This change reads user entered sort type and order by values from advancedsettings.xml.  If the settings aren't in advancedsettings.xml the original defaults are used.

Values are saved in a CViewState class variable allowing for a move to a GUI settings view state option.

If invalid values are entered in either setting it falls back to original defaults.  Duration type was not included to simplify the edit logic and it is an unlikely default.  Size is an option but if the addon doesn't support size, Kodi reverts to name/label as the first sort priority.

## Motivation and Context

The change is motivated by the following

- users don't necessarily want the default sort type or sort order
- users who don't use the default must change it for each folder
- differences from the hard-coded defaults are unnecessarily persisted in the view table
- default assumes users wants to see newest recordings first
- default assumes users record series in air date order

## How Has This Been Tested?

It has been tested on Windows Matrix current nightlies, Estuary skin with the following advancedsettings.xml testing each value

```
  <pvr>
    <pvrrecordings>
      <viewmode>65536</viewmode> <!-- unused  -->
      <sortmethod>2</sortmethod> <!-- 1=Name 2=Date 3=Size 4=File -->
      <sortorder>2</sortorder>  <!-- 1=Ascending 2=Descending -->
      <sortattributes>0</sortattributes> <!-- unused could be watch status -->
    </pvrrecordings>
  </pvr>
```

## Screenshots (if appropriate):

The default value cannot be displayed in Kodi

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [X] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project
- [X] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [X] All new and existing tests passed

### CODE REVIEW

File: xbmc/pvr/windows/GUIViewStatePVR.cpp
Comment (ksooo) [inline xbmc/pvr/windows/GUIViewStatePVR.cpp:55]: As you are only using the 'CViewStet::m_sortDescription' member I see no need to store a 'CViewState' in advancedsettings.xml. This shoul be a 'SortDescription'.

File: xbmc/settings/AdvancedSettings.h
Comment (ksooo) [inline xbmc/settings/AdvancedSettings.h:316]: No need for a 'CViewState' as you are only using the 'SortDescription'.

**Review by ksooo — CHANGES_REQUESTED** [review]

**Comment by ksooo** [comment]
@KarellenX please wait with the Wiki update until this PR got approval.

**Review by ksooo — APPROVED** [review]
lgtm, thx.

File: xbmc/settings/AdvancedSettings.h
Comment (emveepee) [reply xbmc/settings/AdvancedSettings.h]: NP.  The original settings test file is still ok, but now for the wiki  it is 

```
  <pvr>
    <pvrrecordings>
      <sortmethod>2</sortmethod> <!-- 1=Name 2=Date 3=Size 4=File -->
      <sortorder>2</sortorder>  <!-- 1=Ascending 2=Descending -->
    </pvrrecordings>
  </pvr>
```

**Comment by ksooo** [comment]
jenkins build this please

**Comment by DaveTBlake** [comment]
jenkins build this please
@ksooo if happy with this please merge (once jenkins is happy too)

**Comment by phunkyfish** [comment]
I will merge it once Jenkins has passed.

**Comment by phunkyfish** [comment]
Thanks @emveepee!

**Pull request merged**
